toolchain/semantics/semantics_node_kind.def

```diff
@@ -28,6 +28,7 @@ CARBON_SEMANTICS_NODE_KIND(Invalid)
 CARBON_SEMANTICS_NODE_KIND(CrossReference)
 
 CARBON_SEMANTICS_NODE_KIND(Assign)
+CARBON_SEMANTICS_NODE_KIND(Assignment)
```

Comment:
Can we use the same `Assign` node for both the variable declaration case and the assignment statement case? I think that was @jonmeow's intent here.